File: plot/src/main/java/net/fwitz/math/plot/binary/split/SplitComplexFunctionPlot.java
package net.fwitz.math.plot.binary.split;

import net.fwitz.math.complex.SplitComplex;
import net.fwitz.math.plot.binary.BinaryNumberFunctionPlot;
import net.fwitz.math.plot.binary.split.color.DomainColoringSplit;

public class SplitComplexFunctionPlot extends BinaryNumberFunctionPlot<
        SplitComplexFunctionPlot,
        SplitComplexFunctionPanel,
        SplitComplex,
        SplitComplex> {

    public SplitComplexFunctionPlot(String title) {
        super(title);
        colorFn(new DomainColoringSplit());
    }

    @Override
    public SplitComplex value(double x, double y) {
        return SplitComplex.splitComplex(x, y);
    }

    @Override
    protected SplitComplexFunctionPanel createPanel() {
        return new SplitComplexFunctionPanel(this);
    }
}

File: plot/src/main/kotlin/net/fwitz/math/plot/binary/split/SplitComplexFunctionPlot.kt
package net.fwitz.math.plot.binary.split

import net.fwitz.math.binary.split.SplitComplex
import net.fwitz.math.binary.split.SplitComplex.Companion.splitComplex
import net.fwitz.math.plot.binary.BinaryNumberFunctionPlot
import net.fwitz.math.plot.binary.split.color.DomainColoringSplit

class SplitComplexFunctionPlot(
    title: String
) : BinaryNumberFunctionPlot<SplitComplexFunctionPlot, SplitComplexFunctionPanel, SplitComplex, SplitComplex>(
    title
) {
    init {
        colorFn(DomainColoringSplit)
    }

    override fun value(x: Double, y: Double): SplitComplex {
        return splitComplex(x, y)
    }

    override fun createPanel(): SplitComplexFunctionPanel {
        return SplitComplexFunctionPanel(this)
    }
}